projects/aca-shared/rules/src/app.rules.spec.ts

```diff
@@ -793,4 +793,100 @@ describe('app.evaluators', () => {
       expect(app.canOpenWithOffice(context)).toBeTruthy();
     });
   });
+
+  describe('canEditAspects', () => {
+    let context: TestRuleContext;
+    beforeEach(() => {
+      context = new TestRuleContext();
+      context.repository = {
+        version: {
+          major: 10
+        },
+        edition: '',
+        status: undefined
+      } as unknown as RepositoryInfo;
+
+      context.permissions = {
+        check() {
+          return true;
+        }
+      };
+      context.selection.isEmpty = false;
+    });
+
+    it('should return false for multiselection', () => {
+      context.selection.count = 2;
+
+      expect(app.canEditAspects(context)).toBe(false);
+    });
+
+    it('should return false if user cannot update the selected node', () => {
+      context.permissions.check = spyOn(context.permissions, 'check').and.returnValue(false);
+
+      expect(app.canEditAspects(context)).toBe(false);
+    });
+
+    it('should return false if the selected node is write locked', () => {
+      context.selection.file = { entry: { properties: { 'cm:lockType': 'WRITE_LOCK' } } } as NodeEntry;
+
+      expect(app.canEditAspects(context)).toBe(false);
+    });
+
+    it('should return false if the context is trashcan', () => {
+      context.navigation = { url: '/trashcan' };
+
+      expect(app.canEditAspects(context)).toBe(false);
+    });
+
+    it('should return false if the selected node is a smart folder', () => {
+      context.selection.first = { entry: { aspectNames: ['smf:customConfigSmartFolder'], isFolder: true } } as NodeEntry;
+
+      expect(app.canEditAspects(context)).toBe(false);
+    });
+
+    it('should return true if all conditions are met', () => {
+      expect(app.canEditAspects(context)).toBe(true);
+    });
+  });
+
+  describe('canManagePermissions', () => {
+    let context: TestRuleContext;
+    beforeEach(() => {
+      context = new TestRuleContext();
```

Comment:
All that code in beforeEach is the same like in previous describe which you added. Maybe can we create some function which can prepare proper context and reuse it in both beforeEachs?